src/backend/in_memory_record.rs

```diff
@@ -107,14 +108,14 @@ impl<Datum: Clone + PartialEq + Debug, Redeemer: Clone + Eq + PartialEq + Debug
             .collect()
     }
 
-    fn issue(&self, tx: Transaction<Datum, Redeemer>) -> Result<()> {
-        can_spend_inputs(&tx, self.signer.clone())?;
+    fn issue(&self, tx: Transaction<Datum, Redeemer>) -> TxORecordResult<()> {
+        can_spend_inputs(&tx, self.signer.clone())
+            .map_err(|e| TxORecordError::FailedToSpendInputs(Box::new(e)))?;
```

Comment:
This is a design smell on my part. 

`can_spend_inputs` is being used in the `TxORecord` impl, but it's returning a `NauError`.

I think it should probably be moved to the `Backend::process()` method before we call `self.txo_record.issue()`. Feel free to move that. It's kinda out of scope of this story, but then the errors will make sense!